moveit_ros/planning_interface/moveit_cpp/src/planning_component.cpp

```diff
@@ -165,7 +167,8 @@ bool PlanningComponent::plan(const PlanRequestParameters& parameters)
   if (current_goal_constraints_.empty())
   {
     ROS_ERROR_NAMED(LOGNAME, "No goal constraints set for planning request");
-    return false;
+    last_plan_solution_->error_code = MoveItErrorCode(moveit_msgs::MoveItErrorCodes::FAILURE);
```

Comment:
There are meaningful error codes you could use for this. Here I think it should be `INVALID_GOAL_CONSTRAINTS`. The case where joint model group is invalid is basically `INVALID_GROUP_NAME`.